src/Spryker/Zed/AppPayment/Business/AppPaymentFacadeInterface.php

```diff
@@ -27,12 +31,14 @@ interface AppPaymentFacadeInterface
 {
     /**
      * Specification:
+     * - When `PaymentPlatformPluginInterface::initializePayment()` was already executed and the paymentProviderData contains a transaction id (PreOrder payment) the original payment will be added to `InitializePaymentRequestTransfer::payment`.
```

Comment:
I suggest using the existing `InitializePaymentRequestTransfer.orderData.additionalPaymentData` (I originally introduced it for anything payment provider related)

it is mapped directly from Quote https://github.com/spryker/payment/blob/master/src/Spryker/Zed/Payment/PaymentConfig.php#L223

Reply:
It's used in a completely different context and also used as metadata in the StripeIntents. It was also exactly meant for the purpose of transporting additional information of the payment method which should go into the metadata into the Stripe UI.

This datafield is used for controlling the flow. Mixing it up now would require much more code to be implemented to filter data in the metadata field.

But I see what you mean. Let's have a quick call to find the best solution. I also don't like to have too many places where you have or can add data to.

Maybe we can call my field "preOrderPaymentData" and pass it in the "additionalPaymentPaymentData" field. But this would require filtering.